process/transaction/transactionCostEstimator.go

```diff
@@ -44,6 +44,11 @@ func NewTransactionCostEstimator(
 	}, nil
 }
 
+// GasScheduleChange is called when gas schedule is changed, thus all contracts must be updated
+func (tce *transactionCostEstimator) GasScheduleChange(gasSchedule map[string]map[string]uint64) {
+	tce.compilePerByteCost, tce.storePerByteCost = getOperationCost(gasSchedule)
```

Comment:
We need to protect these with a mutex

Reply:
yes. this will be protected as it is accessible through api.

Reply:
protected all the way